### DISCUSSION TITLE
Laravel Backpack saves files twice with wrong directory path

### DISCUSSION THREAD

**Discussion by IlyaBuldakov** [Laravel-Backpack/community-forum#1096] [Q&A (Help)]
I'm using Laravel Backpack.

I have a Product model. It has photos (wow) that I upload when creating a model in the backpack admin panel.

But there is a problem! Photos are uploaded twice:

![image](https://github.com/user-attachments/assets/f78508e2-f032-4652-b597-20178711c163)

Also my database entry:

![image](https://github.com/user-attachments/assets/dd906a4e-8db1-4b3a-9d7e-53ce4e7e4248)

The project is built according to EAV principles and my model has a common setAttribute() method in combination with guarded = [], which allows you to flexibly customize product attributes (they are all different). I mean, the 'img_path' field is also set using this method. I used dd() to see how many times Backpack enters this method. The result is as follows:

![image](https://github.com/user-attachments/assets/6404005c-0e0e-4ace-9aba-200f4b007f89)

Also my model's field `images_directory` wich specifies the path with product's images rewrites by second firing `ProductCrudController::setupCreateOperation() `. Where should I store this field?

Thank you all in advance!

I tried to create my own Uploader, tried to make restrictions in the model. The expected result is that the image is saved once

**Comment by karandatwani92** [comment]
Hey @IlyaBuldakov 

Can you share your model & CRUD controller code? It would help to find & resolve your issue.

**Reply by IlyaBuldakov** [reply]
> Hey @IlyaBuldakov
> 
> Can you share your model & CRUD controller code? It would help to find & resolve your issue.

Hey @karandatwani92,

The problem has been partially solved. It would be good to say for future readers that backpack does not create duplicate files intentionally. 

The error was in the business logic method.

But there is still a problem with the name of the directory where the product photos are stored. The fact is that the `setupCreateOperation()` code fires twice, and each has its own Product model. Because of this, I can't do something like `$imagesDirectory = uniqid()`, because the second time I call `setupCreateOperation()`, this variable will be overwritten.

As I said, it would be nice to write the field to the model. But both times the `setupCreateOperation()` method starts with different "Product" models.

How to solve this problem?

ProductCrudController::setupCreateOperation()

```
protected function setupCreateOperation()
    {
        CRUD::setValidation(ProductRequest::class);

        $product = $this->crud->model;

        CRUD::field('name')->label('Название');
        CRUD::field('price')->label('Цена');
        CRUD::field('description')->label('Описание');

        $imagesDirectory = $this->crud->get('images_directory');

        CRUD::field('images_directory')->type('hidden')->value($imagesDirectory);

        CRUD::field('img_path')->label('Изображения (от 1 до 6)')->type('upload_multiple')->withFiles([
            'disk' => 'product_images',
            'path' => $imagesDirectory
        ]);

        $category = $this->crud->getRequest()->get('category');

        CRUD::field('category_id')->type('hidden')->value($category);

        $product->category = $category;

        if ($category) {
            $keys = $this->keyService->getAllByCategoryId($category);

            foreach ($keys as $key) {
                CRUD::addField([
                    'name' => $key->name,
                    'label' => $key->alias
                ]);
            }
        }
    }
```

**Comment by IlyaBuldakov** [comment]
I have edited my question so that desperate seekers can find it in case of one of these problems.

**Comment by altayevrim** [comment]
I have the same issue. I think this should be considered as a bug.

**Reply by altayevrim** [reply]
I think I found the issue. I was passing user id with created event on model itself due to my laziness to check Backpack docs.
https://shottr.cc/s/1LR5/SCR-20240801-hne.png

This makes the file to be uploaded twice.
